smarts/core/sumo_traffic_simulation.py

```diff
@@ -546,9 +541,8 @@ def _sync(self, provider_state: ProviderState):
             self._traci_conn.vehicle.setSpeedMode(vehicle_id, all_checks)
             self._traci_conn.vehicle.setSpeed(vehicle_id, -1)
 
-        if self._endless_traffic:
-            self._reroute_vehicles(traffic_vehicle_states)
-            self._teleport_exited_vehicles()
+        self._reroute_vehicles(traffic_vehicle_states)
+        self._teleport_exited_vehicles()
```

Comment:
This calls the `traci_conn.simulation.getSubscriptionResults().` every `sync(...)` in addition to the other time it is called in `step(...)`. If possible it would be best if we only ever do that call once.

Reply:
Good call!  I just pushed something to avoid calling that more than once.